hnswlib/hnswalg.h

```diff
@@ -977,15 +955,15 @@ namespace hnswlib {
             return result;
         };
 
-        tableint addPoint(const void *data_point, labeltype label, int level) {
+        tableint addPoint(const void *data_point, labeltype label, int level, bool bUpdate = false) {
 
             tableint cur_c = 0;
             {
                 // Checking if the element with the same label already exists
                 // if so, updating it *instead* of creating a new element.
                 std::unique_lock <std::mutex> templock_curr(cur_element_count_guard_);
                 auto search = label_lookup_.find(label);
-                if (search != label_lookup_.end()) {
+                if (bUpdate && search != label_lookup_.end()) {
```

Comment:
I do not understand the `bUpdate` logic here. It add a new element with the same label? What happens to the first element? How `label_lookup_` should operate?

Reply:
From sfit_1b.cpp example, it seems that you sometimes use 'label' as the unique ID of elements (when Building index), sometimes use 'label' as the class of elements (see the usage of the return value of searchKnn). I think 'lable' should mean class. For example, in MNIST its value is one of (0,1,2,3,4,5,6,7,8,9). Then, the reason of use 'bUpdate' is obviously. We don't want to update an existing class every time we add a new point. We only let this happen when we need (e.g at online training). At that time, the data of the closest node will be replaced by the new coming data.
If this is not your original intention, omit this suggestion.

Reply:
But the question is where and how to find the class of the elements?

Reply:
I think the class labels should be stored by the user (in a dict or an array)
From a design point of view, there should be a mechanism for the library user to extract the vectors back/delete/update by using a key. There are no other keys defined in the library so we ended up use labels as keys. To change that there should be a good alternative to support those operations.

Reply:
I don't understand, it is hard to add an 'ID' at data_level0_memory_?

Reply:
If the labels are stored in a dict or an array, it still need an algorithm to retrieve them.

Reply:
It is possible, but I think it will break some stuff like compatibility with old saved indices as this label also needs to be stored somewhere.
It also would add complexity to the interface, there would be both id and and class in knn. 
While labels for classifications can be done from the user side, both in C++ and python via simple lookups.